tasks/ssh.yml

```diff
@@ -41,23 +44,37 @@
     - __kdump_authorized_keys_file.stat.isreg is defined
     - __kdump_authorized_keys_file.stat.isreg
 
-- name: Write new authorized_keys if needed
+- name: Handle authorized_keys update if needed
   vars:
     # note - Cg== is the newline character in base64
     __kdump_authorized_keys_lines: "{{
       (__kdump_authorized_keys.content | d('Cg==') | b64decode).split('\n') |
       reject('match', '^$') | list }}"
     __kdump_new_key: "{{ keydata.content | b64decode | trim }}"
-  copy:
-    content: "{{ ((__kdump_authorized_keys_lines + [__kdump_new_key]) |
-      join('\n')) ~ '\n' }}"
-    dest: "{{ __kdump_authorized_keys_file.stat.path |
-              d(__kdump_authorized_keys_path) }}"
-    group: "{{ __kdump_authorized_keys_file.stat.group | d(kdump_ssh_user) }}"
-    owner: "{{ __kdump_authorized_keys_file.stat.owner | d(kdump_ssh_user) }}"
-    mode: "{{ __kdump_authorized_keys_file.stat.mode | d('0600') }}"
-  delegate_to: "{{ kdump_ssh_server }}"
   when: not __kdump_new_key in __kdump_authorized_keys_lines
+  delegate_to: "{{ kdump_ssh_server }}"
+  block:
+    - name: Ensure ssh directory for authorized_keys if needed
+      file:
+        path: "{{ __kdump_ssh_path }}"
+        state: directory
+        group: "{{ __kdump_authorized_keys_file.stat.group |
+          d(kdump_ssh_user) }}"
+        owner: "{{ __kdump_authorized_keys_file.stat.owner |
+          d(kdump_ssh_user) }}"
+        mode: "{{ __kdump_authorized_keys_file.stat.mode | d('0700') }}"
```

Comment:
If we look at the __kdump_authorized_keys_file.stat.mode, which is going to generally be "600", I think this will cause scenarios where the .ssh directory ends up with "600" permissions (which will cause the user to not have access to the directory at all).   In trying out this PR this morning, I had this happen once -- but I'm not exactly sure what exact steps lead up to this happening.

Reply:
It appears that on subsequent runs of the role, if additional keys are being added to authorized_keys, this block is run, and the .ssh directory is changed to 600 mode, making it unreadable.

Reply:
fixed

I also discovered that `stat.owner` should be `stat.pw_name`, and `stat.group` should be `stat.gr_name`